tensorflow/security/fuzzing/cc/fuzz_session.h

```diff
@@ -64,18 +64,30 @@ namespace fuzzing {
 // Used by GFT to map a known domain (vector<T>) to an unknown
 // domain (Tensor of datatype). T and datatype should match/be compatible.
 template <typename T = uint8_t>
-inline auto AnyTensor() {
+inline auto AnyTensor(const int rank = -1) {
+  T min_size{1};
+  T max_size{TensorShape::MaxDimensions()};
+  if (rank != -1) {
+    min_size = max_size = rank;
+  }
   return fuzztest::Map(
-      [](auto v) {
-        Tensor tensor(DataTypeToEnum<T>::v(),
-                      TensorShape({static_cast<int64_t>(v.size())}));
+      [](auto t) {
+        std::vector<int64_t> shape;
+        shape.reserve(t.size());
+        std::transform(t.cbegin(), t.cend(), std::back_inserter(shape),
+            [](auto& dim) { return static_cast<int64_t>(dim.size()); });
+
+        Tensor tensor(DataTypeToEnum<T>::v(), TensorShape(shape));
```

Comment:
Should this construct only valid `TensorShape`s?

Reply:
@mihaimaruseac it's a good question!
@fcoUnda  what do you think, should we generate valid Tensor's shapes only or is it good to have invalid ones?
It is question about approach: should we have separate dedicated fuzzer for Tensor/TensorShape or is it good to have also fuzzer that covers both Tensor/TensorShape and Operation itself (kind of integration fuzzer)? Maybe this "integration" fuzzer could reveal some hidden behavior?

Reply:
There are way more invalid shapes than valid ones, I'd assume fuzzer would be stuck with invalid ones for no real benefit.

Reply:
Now that we have flatmap, this should be much easier to modularize. I propose creating a function that creates shape domains. We can start with only valid domains, but it could also have an  boolean argument for the two options.

Reply:
I forgot to elaborate: then we can use flatmap to create tensors of any (or fixed shapes) using the above method.

Reply:
done